### PR TITLE
[IOTDB-873] Add count devices DDL

### PR DESCRIPTION
**Pull request by yhwang-hbl**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/MTree.java [hunk 1]
```diff
@@ -695,6 +708,38 @@ private int getCount(MNode node, String[] nodes, int idx) throws MetadataExcepti
...
+        }
+      }
+    } else {
+      boolean deviceAdded = false;
+      for (MNode child : node.getChildren().values()) {
+        if (child instanceof MeasurementMNode ) {
+          if( !deviceAdded){
+            cnt++;
+            deviceAdded = true;
+          }else{
```
Comment (neuyilan) [inline server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:732]: code format
}else{- > } else {
if( !deviceAdded){. -> if( !deviceAdded) {

please use google style format: http://iotdb.apache.org/zh/Development/ContributeGuide.html

**Review by neuyilan — CHANGES_REQUESTED** [review]

**Review by samperson1997 — APPROVED** [review]
Hi, everything LGTM except code format ; )

File: server/src/main/java/org/apache/iotdb/db/metadata/MTree.java (hunk 1 above)
Comment (yhwang-hbl) [reply server/src/main/java/org/apache/iotdb/db/metadata/MTree.java]: i got this

**Comment by yhwang-hbl** [comment]
> Hi, everything LGTM except code format ; )

Thank you. I adjusted it

**Review by neuyilan — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/metadata/MTree.java [hunk 2]
```diff
@@ -695,6 +708,38 @@ private int getCount(MNode node, String[] nodes, int idx) throws MetadataExcepti
...
+          if (!deviceAdded) {
+            cnt++;
+            deviceAdded = true;
+          } else {
+            break;
+          }
+        } else {
+          cnt += getDevicesCount(child, nodes, idx + 1);
+        }
+      }
```
Comment (Alima777) [inline server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:738]: Hi, we support subdevice like "root.sg.device1.subdevice1.sensor1" while subdevice1 is a measurementNode and a device. In your case, the subdevice1 will be ignored.

You can refer to the Mtree.findDevice() method.

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBMetadataFetchIT.java [hunk 3]
```diff
@@ -53,7 +56,7 @@ private static void insertSQL() throws ClassNotFoundException, SQLException {
         statement.execute(sql);
       }
     } catch (Exception e) {
-      e.printStackTrace();
+      logger.error("insertSQL failed", e);
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBMetadataFetchIT.java:59]:
```suggestion
      logger.error("insertSQL() failed", e);
```

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBMetadataFetchIT.java [hunk 4]
```diff
@@ -118,7 +121,7 @@ public void showTimeseriesTest() throws ClassNotFoundException, SQLException {
           }
           Assert.assertEquals(standard, builder.toString());
         } catch (SQLException e) {
-          e.printStackTrace();
+          logger.error("showTimeseriesTest failed", e);
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBMetadataFetchIT.java:124]:
```suggestion
          logger.error("showTimeseriesTest() failed", e);
```

Others same :D

**Review by Alima777 — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/apache/iotdb/db/metadata/MTree.java (hunk 2 above)
Comment (samperson1997) [reply server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:738]: Ah.. good point. Maybe @yhwang-hbl could add a specific test case for this example ; )

Comment (yhwang-hbl) [reply server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:738]: > Hi, we support subdevice like "root.sg.device1.subdevice1.sensor1" while subdevice1 is a measurementNode and a device. In your case, the subdevice1 will be ignored.
> 
> You can refer to the Mtree.findDevice() method.
hi, I didn't know the concept of sub devices before. I did a self-test on my code, which will filter out the situation you said. I will improve it,thanks for your guidance

Comment (yhwang-hbl) [reply server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:738]: > Ah.. good point. Maybe @yhwang-hbl could add a specific test case for this example ; )

OK，i will improve it and add a test for the corresponding situation

**Review by Alima777 — APPROVED** [review]
It looks good to me now~ Thank you for quick fix  :D

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**